src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -0,0 +1,91 @@
+package xerus.monstercat.api
+
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import mu.KotlinLogging
+import xerus.ktutil.javafx.properties.SimpleObservable
+import xerus.ktutil.javafx.properties.bindSoft
+import xerus.monstercat.api.response.Track
+import java.util.*
+import kotlin.collections.ArrayList
+
+object Playlist {
+	val logger = KotlinLogging.logger { }
+	
+	val tracks: ObservableList<Track> = FXCollections.observableArrayList()
+	val history = ArrayDeque<Track>()
+	val currentIndex = SimpleObservable<Int?>(null).apply {
+		bindSoft({
+			tracks.indexOf(Player.activeTrack.value).takeUnless { i -> i == -1 }
+		}, Player.activeTrack, tracks)
+	}
+	
+	var repeat = false
+	var shuffle = false
+	
+	operator fun get(index: Int): Track? = tracks.getOrNull(index)
+	
+	/** This property prevents songs that come from the [history] from being added to it again,
+	 * which would effectively create an infinite loop */
+	var lastPolled: Track? = null
+	
+	fun getPrev(): Track? = history.pollLast()?.also {
+		lastPolled = it
+		logger.debug("Polled $it from History")
+	} ?: currentIndex.value?.let { get(it - 1) }
+	
+	fun getNext() : Track?{
+		return when {
```

Comment:
You can use `=` syntax here

Reply:
363824a8dbe724ce172257b20ab8b1548b5378cf